pkg/servicelb/controller.go

```diff
@@ -206,8 +212,14 @@ func serviceIPs(svc *core.Service) []string {
 	return ips
 }
 
+// podIPs returns a list of IPs for Nodes hosting ServiceLB Pods.
+// If at least one node has External IPs available, only external IPs are returned.
+// If no nodes have External IPs set, the Internal IPs of all nodes running pods are returned.
 func (h *handler) podIPs(pods []*core.Pod) ([]string, error) {
```

Comment:
I'm curious if we could consolidate this function. Here is a small pseudocode version...

```
for pod in pods {
    // get node
    for addr in node.addrs {
        // perform keys function here and dynamically determine internal or external
    }
}
// rootless check, etc.
return ips, nil
```

Reply:
I considered that, but I think I would prefer to not mix ExternalIP and InternalIPs, as that could lead to some inconsistent behavior if nodes have mixed connectivity. See the discussion at https://github.com/k3s-io/k3s/pull/2909#issuecomment-775538812